ui/StatusQ/include/StatusQ/keychain.h

```diff
@@ -61,6 +62,7 @@ class Keychain : public QObject {
 #ifdef Q_OS_MACOS
     Status saveCredential(const QString &account, const QString &password);
     Status deleteCredential(const QString &account);
-    Status getCredential(const QString &account, QString *out);
+    Status getCredential(const QString &reason, const QString &account, QString *out);
```

Comment:
```suggestion
    Status getCredential(const QString &reason, const QString &account, QString *out) const;
```

Reply:
`getCredential` changes `m_activeAuthContext`, and I don't think it makes much sense to make it mutable 🤔

Reply:
The general rule is that method should be `const` if the state of the object observed by the caller stays intact after calling that method. It's the case there. So using `mutable` keyword for `m_activeAuthContext` would be justified there in the end.

Reply:
Ok, it's private method...